lib/rubocop/cop/performance/count.rb

```diff
@@ -87,7 +87,7 @@ def autocorrect(corrector, node, selector_node, selector)
         end
 
         def eligible_node?(node)
-          !(node.parent && node.parent.block_type?)
+          !(node.parent && node.parent.block_type?) && (node.receiver.call_type? || node.receiver.body)
```

Comment:
The condition seems a bit complex. How about refactoring it as follows?

```suggestion
          return false if node.parent&.block_type?

          node.receiver.call_type? || node.receiver.body
```